src/index.js

```diff
@@ -6,13 +6,14 @@ var util = require('util')
 function Walker (dir, options) {
   assert.strictEqual(typeof dir, 'string', '`dir` parameter should be of type string. Got type: ' + typeof dir)
   var defaultStreamOptions = { objectMode: true }
-  var defaultOpts = { queueMethod: 'shift', pathSorter: undefined, filter: undefined }
+  var defaultOpts = { queueMethod: 'shift', pathSorter: undefined, filter: undefined, depthLimit: undefined }
   options = Object.assign(defaultOpts, options, defaultStreamOptions)
 
   Readable.call(this, options)
   this.root = path.resolve(dir)
   this.paths = [this.root]
   this.options = options
+  if (options.depthLimit > -1) this.rootDepth = path.normalize(this.root).split(path.sep).length + 1
```

Comment:
@gsf-sellis no need to use `path.normalize()` here since `path.resolve()` normalizes the resulting path.